### PR TITLE
feat(deploy/ha): Add ability to delete orphan services.

### PR DESCRIPTION
**Pull request by joonlim**

### CODE REVIEW

**Comment by joonlim** [comment]
@jtk54 @lwander

File: docs/commands.md [hunk 1]
```diff
@@ -7749,6 +7749,7 @@ hal deploy apply [parameters]
 
 #### Parameters
  * `--auto-run`: This command will generate a script to be run on your behalf. By default, the script will run without intervention - if you want to override this, provide "true" or "false" to this flag.
+ * `--delete-orphan-services`: (*Default*: `false`) Deletes unused Spinnaker services after the deploy succeeds.
```
Comment (lwander) [inline docs/commands.md]: Same comment here about "orphaned" services

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/KubectlDeployer.java [hunk 2]
```diff
@@ -122,4 +122,36 @@ public void flushInfrastructureCaches(KubectlServiceProvider serviceProvider,
...
+      if (service instanceof SidecarService) {
+        return;
+      }
+
+      ServiceSettings settings = resolvedConfiguration.getServiceSettings((SpinnakerService) service);
+      if (settings == null) {
+        return;
+      }
+
+      if (settings.getEnabled() == null || settings.getEnabled()) {
```
Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/KubectlDeployer.java]: Can you verify that this doesn't delete a service that was omitted by passing in `--service-names` for a deployment?

File: docs/commands.md (hunk 1 above)
Comment (jtk54) [reply docs/commands.md]: :+1:

File: halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/deploy/ApplyDeployCommand.java [hunk 3]
```diff
@@ -84,16 +84,24 @@
   )
   List<String> excludeServiceNames = new ArrayList<>();
 
+  @Parameter(
+      names = "--delete-orphan-services",
```
Comment (jtk54) [inline halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/deploy/ApplyDeployCommand.java]: s/orphan/orphaned/

Comment (jtk54) [reply halyard-cli/src/main/java/com/netflix/spinnaker/halyard/cli/command/v1/deploy/ApplyDeployCommand.java]: I don't know if you want to change the variable names as well here, matters less than the user facing stuff.

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/KubectlDeployer.java (hunk 2 above)
Comment (joonlim) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/KubectlDeployer.java]: verified

Comment (joonlim) [reply halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/deployment/v1/KubectlDeployer.java]: Services provided in the service-names flag will be used in a list of services to be examined. The enabled flag is set/not set by the ServiceProvider later on.

**Pull request merged**